src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1132,8 +1167,15 @@ private void AddTransactionToWallet(Transaction transaction, TxOut utxo, int? bl
 
                 if (isPropagated)
                     foundTransaction.IsPropagated = true;
+
+                if (block != null)
```

Comment:
...and here if the transaction is part of a block, you remove the corresponding TransactionData one from the index.